src/main/java/org/threadly/litesockets/SocketExecuterCommonBase.java

```diff
@@ -23,45 +23,39 @@
  *  This is a common base class for the Threaded and NoThread SocketExecuters. 
  */
 abstract class SocketExecuterCommonBase extends AbstractService implements SocketExecuter {
-  private final Logger log = Logger.getLogger(this.getClass().toString());
-  protected final SubmitterScheduler acceptScheduler;
-  protected final SubmitterScheduler readScheduler;
-  protected final SubmitterScheduler writeScheduler;
+  protected final Logger log = Logger.getLogger(this.getClass().toString());
   protected final SubmitterScheduler schedulerPool;
+  protected final SubmitterScheduler acceptScheduler;
   protected final ConcurrentHashMap<SocketChannel, Client> clients = new ConcurrentHashMap<>();
   protected final ConcurrentHashMap<SelectableChannel, Server> servers = new ConcurrentHashMap<>();
   protected final SocketExecuterByteStats stats = new SocketExecuterByteStats();
   protected final WatchdogCache dogCache;
-  protected volatile boolean verboseLogging = false;
-  protected Selector readSelector;
-  protected Selector writeSelector;
+  protected volatile boolean perConnectionStatsEnabled = true;
   protected Selector acceptSelector;
 
   SocketExecuterCommonBase(final SubmitterScheduler scheduler) {
-    this(scheduler,scheduler,scheduler,scheduler);
+    this(scheduler, scheduler);
   }
 
-  SocketExecuterCommonBase(final SubmitterScheduler acceptScheduler, 
-      final SubmitterScheduler readScheduler, 
-      final SubmitterScheduler writeScheduler, 
-      final SubmitterScheduler ssi) {
+  SocketExecuterCommonBase(final SubmitterScheduler acceptScheduler, final SubmitterScheduler ssi) {
     log.setParent(Logger.getGlobal());
     ArgumentVerifier.assertNotNull(ssi, "ThreadScheduler");    
     ArgumentVerifier.assertNotNull(acceptScheduler, "Accept Scheduler");
-    ArgumentVerifier.assertNotNull(readScheduler, "Read Scheduler");
-    ArgumentVerifier.assertNotNull(writeScheduler, "Write Scheduler");
+    
     schedulerPool = ssi;
     dogCache = new WatchdogCache(ssi, true);
     this.acceptScheduler = acceptScheduler;
-    this.readScheduler = readScheduler;
-    this.writeScheduler = writeScheduler;
   }
 
-  protected void addReadAmount(int size) {
+  public void setPerConnectionStatsEnabled(boolean enabled) {
```

Comment:
Not sure what you think about this, but it seems like we should loop through all clients and enable or disable them when this changes.

Reply:
I considered looping through, it would require locking to be 100%, so I was thinking that this is probably good enough.  Let me know your thoughts

Reply:
You could do it with schedule on the selector threads, not sure if its worth it or not, just an idea.

Reply:
For now I say we leave it, I can't imagine changing this while clients are active is likely